chainer/optimizer.py

```diff
@@ -308,9 +310,7 @@ def update_core_chainerx(self, param):
         temp_param = variable.Variable._init_unchecked(
             param._chainerx_fallback_array, is_chainerx_array=False)
 
-        if grad_array is not None:
```

Comment:
I also refactored here, but I'm not sure what kind of case is considered originally.
Please review.